tools/internal_ci/helper_scripts/move_src_tree_and_respawn_itself_rc

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Source this rc script at the beginning of your CI script to move
+# the entire source tree to a different directory.
+# That avoids slow finalization (and a number of other problems)
+# caused by kokoro rsyncing the entire /tmpfs/src directory
+# back to the agent.
+# See b/74837748 for more context.
+
+set -ex
+
+CURRENT_DIR="$(pwd)"
+CI_SCRIPT_RELATIVE_TO_SRC="$0"
```

Comment:
Good to learn the "$0" of sourced script still points to the parent file.

Reply:
it kind of makes sense, since "source" acts as if you just included the contents of the script into the current script.